### ISSUE TITLE
Add --regex-limit

### ISSUE THREAD

**Issue by swissspidy** [wp-cli/search-replace-command#61]
When using the `search-replace` command with the `--regex` option, it uses `preg_replace()` / `preg_replace_callback()` to replace all occurrences in a row.

There might be [situations](https://wordpress.stackexchange.com/questions/291706/using-wp-cli-and-delete-everypost-first-image) where you only want that replacement to happen a certain amount of times (e.g. only once per row).

`preg_replace_callback()` and `preg_replace()` have a `$limit` parameter that could be used for this and defaults to `-1` (no limit).

Suggestion: Allow modifying this `$limit` parameter using a new `--regex-limit` argument. That should be rather trivial (just pass the value to the function), but needs some testing obviously.

**Comment by gitlost** [comment]
Note @swissspidy you can actually do the case mentioned in the WordPress StackExchange question you reference with a regex, eg:
```
wp search-replace '^((?:(?!(?+1)).)*+)(<img[^>]+>)' '\1` wp_posts --regex --regex-flags=is
```
(variation of MrBull's note http://php.net/manual/en/function.preg-match.php#103015), though obviously having a `--regex-limit` option is much handier/better.

Actually first used a non-greedy `*?` here instead of a possessive `*+` which lead to a `PREG_JIT_STACKLIMIT_ERROR`  on the `preg_replace()` at [SearchReplacer.php#L105](https://github.com/wp-cli/search-replace-command/blob/v1.2.0/src/WP_CLI/SearchReplacer.php#L105) due to data size in the test WP install I was using. So this is a bug - need to check for a `null` return here - will do an issue for it.

**Comment by CodeProKid** [comment]
Seems like this can be closed as it was fixed in #62

**Comment by schlessera** [comment]
Fixed via #62
